weed/storage/volume_vacuum.go

```diff
@@ -327,7 +341,11 @@ func (v *Volume) makeupDiff(newDatFileName, newIdxFileName, oldDatFileName, oldI
 		}
 	}
 
-	return nil
+	if v.needleMapKind == NeedleMapInMemory {
+		return v.tmpNm.DoOffsetLoading(nil, idx, uint64(idxSize)/types.NeedleMapEntrySize)
```

Comment:
I am confused by the logic. It is reading a new idx file from its end, and then process the index entries. So it will read zero index entries?